### ISSUE TITLE
Revise grammar and syntax model for `scoped`

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#62983]
TL;DR: From LDM 7/27, we should write the language spec grammar for ref locals, ref fields, and `scoped`. Then we should consider how that affects our syntax model.

LDM expressed the desire to not put `scoped` into modifiers buckets, as modifiers can generally be re-ordered.
Instead `scoped` should be tied to the type. One proposal is to introduce `type_modifiers` following `modifiers` to hold `scoped`, `ref` and `readonly` in specific configurations.
This should take inspiration from Rex's spec draft for ref locals: https://github.com/dotnet/csharpstandard/pull/213

Once that is done, we need to decide how to adjust our implementation.
1. do we want to keep `scoped` in modifiers, or can we stick it into `RefType` (extending its meaning to "type with type modifiers"), or maybe introduce a new `ScopedType`?
2. do we want to make use of that syntax model for new scenarios in parameters? (parameters don't currently make use of `RefType` and instead stick everything into modifiers for backcompat reasons)

Relates to test plan https://github.com/dotnet/roslyn/issues/59194
